Lib/textwrap.py

```diff
@@ -218,8 +218,10 @@ def _handle_long_word(self, reversed_chunks, cur_line, cur_len, width):
             end = space_left
             chunk = reversed_chunks[-1]
             if self.break_on_hyphens and len(chunk) > space_left:
+                # break after last hyphen, but only if there are
+                # non-hyphens before it
                 hyphen = chunk.rfind('-', 0, space_left)
-                if hyphen != -1:
+                if hyphen > 0 and any(c != '-' for c in chunk[0:hyphen]):
```

Comment:
Simply `chunk[:hyphen]` instead of `chunk[0:hyphen]`.

Other ways to perform this test are:
```python
chunk[:hyphen].rstrip('-')
```
and
```python
re.compile(r'[^-]').search(chunk, 0, hyphen)
```
What do you prefer?

Reply:
I chose this method on the unsubstantiated hunch that it would have better performance - rstrip creates a new string, right? And re seemed like overkill. But I don't feel strongly about it.

Yes, the 0 index can be dropped.

Reply:
Slicing also creates a new string.

Your version looks the most obvious (it may the slowest one because Python generators are slower than C code), strip-version is the shortest code, and re-version may be potentially the most efficient for very long strings, but I am not sure that the difference exist in common case.

Reply:
I find the rstrip version most readable, so if there is no performance concern I would go with that.

Reply:
Well, it does not matter and I already merged your initial version. Thank you for your PR!

Reply:
Agreed, thank you for the review and merge!